File: java/oop/AccountPolymorphism/src/com/techlabs/accountpoly/SavingsAccount.java
package com.techlabs.accountpoly;

public class SavingsAccount extends Account{
	public SavingsAccount(int accNo,String name,double balance){
		super(accNo,name,balance);
	}

	@Override
	public void withdraw(double amount) {
		if(validateWithdraw(amount)){
			this.balance-=amount;
		}
		else{
			System.out.println("Cannot Withdraw Minimum Balance has to be maintained to 500 Rs");
		}
	}
	
	private boolean validateWithdraw(double amount){
		if((this.balance-amount)<500)
			return false;
		return true;
	}
}

File: dotNet/CSharp/AccountAppPoly/AccountAppPoly/SavingsAccount.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace AccountAppPoly
{
    class SavingsAccount : Account
    {
        public SavingsAccount(int accNo, string name, double balance) : base(accNo, name, balance)
        {
        }

        public override void Withdraw(double amount)
        {
            if (ValidateWithdraw(amount))
            {
                this.Balance -= amount;
            }
            else
            {
                Console.WriteLine("Cannot Withdraw Minimum Balance has to be maintained to 500 Rs");
            }
        }

        private Boolean ValidateWithdraw(double amount)
        {
            if ((this.Balance-amount) < 500)
            {
                return false;
            }
            return true;
        }
    }
}
